server/src/main/java/org/elasticsearch/index/query/IntervalQueryBuilder.java

```diff
@@ -0,0 +1,144 @@
+    }
+
+    @Override
+    protected void doWriteTo(StreamOutput out) throws IOException {
+        out.writeString(field);
+        out.writeNamedWriteable(sourceProvider);
+    }
+
+    @Override
+    protected void doXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject(NAME);
+        builder.field(field);
+        builder.startObject();
+        sourceProvider.toXContent(builder, params);
+        printBoostAndQueryName(builder);
+        builder.endObject();
+        builder.endObject();
+    }
+
+    public static IntervalQueryBuilder fromXContent(XContentParser parser) throws IOException {
+        if (parser.nextToken() != XContentParser.Token.FIELD_NAME) {
+            throw new ParsingException(parser.getTokenLocation(), "Expected [FIELD_NAME] but got [" + parser.currentToken() + "]");
+        }
+        String field = parser.currentName();
+        if (parser.nextToken() != XContentParser.Token.START_OBJECT) {
+            throw new ParsingException(parser.getTokenLocation(), "Expected [START_OBJECT] but got [" + parser.currentToken() + "]");
+        }
+        String name = null;
+        float boost = 1;
+        IntervalsSourceProvider provider = null;
+        while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
+            if (parser.currentToken() != XContentParser.Token.FIELD_NAME) {
+                throw new ParsingException(parser.getTokenLocation(),
+                    "Expected [FIELD_NAME] but got [" + parser.currentToken() + "]");
+            }
+            switch (parser.currentName()) {
+                case "_name":
+                    parser.nextToken();
+                    name = parser.text();
+                    break;
+                case "boost":
+                    parser.nextToken();
+                    boost = parser.floatValue();
+                    break;
+                default:
+                    provider = IntervalsSourceProvider.fromXContent(parser);
+
+            }
+        }
+        if (parser.nextToken() != XContentParser.Token.END_OBJECT) {
+            throw new ParsingException(parser.getTokenLocation(),
+                "Expected [END_OBJECT] but got [" + parser.currentToken() + "]");
+        }
+        if (provider == null) {
+            throw new ParsingException(parser.getTokenLocation(), "Missing intervals from interval query definition");
+        }
+        IntervalQueryBuilder builder = new IntervalQueryBuilder(field, provider);
+        builder.queryName(name);
+        builder.boost(boost);
+        return builder;
+
+    }
+
+    @Override
+    protected Query doToQuery(QueryShardContext context) throws IOException {
+        MappedFieldType fieldType = context.fieldMapper(field);
+        if (fieldType == null) {
+            throw new IllegalArgumentException("Cannot create IntervalQuery over non-existent field [" + field + "]");
+        }
```

Comment:
in general we are lenient with unmapped fields because of cross-index search, should we be lenient here too?

Reply:
Have updated